datasets/mlsum/mlsum.py

```diff
@@ -0,0 +1,133 @@
+    VERSION = nlp.Version("1.0.0")
+    BUILDER_CONFIGS = (
+        [
+            MlsumConfig(
+                name=lang,
+                description="",
+            )
+            for lang in _LANG
+        ]
+    )
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # nlp.features.FeatureConnectors
+
+            features=nlp.Features(
+                {
+                    "text": nlp.Value("string"),
+                    "summary": nlp.Value("string"),
+                    "topic": nlp.Value("string"),
+                    "url": nlp.Value("string"),
+                    "title": nlp.Value("string"),
+                    "date":nlp.Value("string")
+                    # These are the features of your dataset like images, labels ...
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # dl_manager is a nlp.download.DownloadManager that can be used to
+        # download and extract URLs
+        
+        lang = str(self.config.name)
+        urls_to_download = {
+            "test": os.path.join(_URL, lang+"_test.zip"),
+            "train": os.path.join(_URL, lang+"_train.zip"),
+            "validation": os.path.join(_URL, lang+"_val.zip")
+        }
+        downloaded_files = dl_manager.download_and_extract(urls_to_download)
+
+        return [
+            nlp.SplitGenerator(
+                name=nlp.Split.TRAIN,
+                    # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(downloaded_files["train"], lang+'_train.jsonl'),
+                    "lang": lang,
+                },
+            ),
+            nlp.SplitGenerator(
+                name=nlp.Split.VALIDATION,
+                    # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(downloaded_files["validation"], lang+'_val.jsonl'),
+                    "lang": lang,
+                },
+            ),
+            nlp.SplitGenerator(
+                name=nlp.Split.TEST,
+                    # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(downloaded_files["test"], lang+'_test.jsonl'),
+                    "lang": lang,
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath, lang):
+        """Yields examples."""
+        with open(filepath) as f:
```

Comment:
To fix the CI you have to specify an encoding here (ex: `encoding="utf-8"`).
This is for windows users that may encounter issues with some codecs